File: typescript/secret-handshake/secret-handshake.test.ts
import { commands } from './secret-handshake'

describe('Secret Handshake', () => {
  describe('Create A Handshake For A Number', () => {
    it('wink for 1', () => {
      expect(commands(1)).toEqual(['wink'])
    })

    it('double blink for 10', () => {
      expect(commands(2)).toEqual(['double blink'])
    })

    it('close your eyes for 100', () => {
      expect(commands(4)).toEqual(['close your eyes'])
    })

    it('jump for 1000', () => {
      expect(commands(8)).toEqual(['jump'])
    })

    it('combine two actions', () => {
      expect(commands(3)).toEqual(['wink', 'double blink'])
    })

    it('reverse two actions', () => {
      expect(commands(19)).toEqual(['double blink', 'wink'])
    })

    it('reversing one action gives the same action', () => {
      expect(commands(24)).toEqual(['jump'])
    })

    it('reversing no actions still gives no actions', () => {
      expect(commands(16)).toEqual([])
    })

    it('all possible actions', () => {
      expect(commands(15)).toEqual([
        'wink',
        'double blink',
        'close your eyes',
        'jump',
      ])
    })

    it('reverse all possible actions', () => {
      expect(commands(31)).toEqual([
        'jump',
        'close your eyes',
        'double blink',
        'wink',
      ])
    })

    it('do nothing for zero', () => {
      expect(commands(0)).toEqual([])
    })
  })
})

File: dart/secret-handshake/test/secret_handshake_test.dart
import 'package:secret_handshake/secret_handshake.dart';
import 'package:test/test.dart';

void main() {
  final secretHandshake = SecretHandshake();

  group('SecretHandshake', () {
    test('wink for 1', () {
      final result = secretHandshake.commands(1);
      expect(result, equals(<String>['wink']));
    }, skip: false);

    test('double blink for 10', () {
      final result = secretHandshake.commands(2);
      expect(result, equals(<String>['double blink']));
    }, skip: false);

    test('close your eyes for 100', () {
      final result = secretHandshake.commands(4);
      expect(result, equals(<String>['close your eyes']));
    }, skip: false);

    test('jump for 1000', () {
      final result = secretHandshake.commands(8);
      expect(result, equals(<String>['jump']));
    }, skip: false);

    test('combine two actions', () {
      final result = secretHandshake.commands(3);
      expect(result, equals(<String>['wink', 'double blink']));
    }, skip: false);

    test('reverse two actions', () {
      final result = secretHandshake.commands(19);
      expect(result, equals(<String>['double blink', 'wink']));
    }, skip: false);

    test('reversing one action gives the same action', () {
      final result = secretHandshake.commands(24);
      expect(result, equals(<String>['jump']));
    }, skip: false);

    test('reversing no actions still gives no actions', () {
      final result = secretHandshake.commands(16);
      expect(result, equals(<String>[]));
    }, skip: false);

    test('all possible actions', () {
      final result = secretHandshake.commands(15);
      expect(result, equals(<String>['wink', 'double blink', 'close your eyes', 'jump']));
    }, skip: false);

    test('reverse all possible actions', () {
      final result = secretHandshake.commands(31);
      expect(result, equals(<String>['jump', 'close your eyes', 'double blink', 'wink']));
    }, skip: false);

    test('do nothing for zero', () {
      final result = secretHandshake.commands(0);
      expect(result, equals(<String>[]));
    }, skip: false);
  });
}
